### PR TITLE
Helm Chart for the csi-secrets-store-provider-aws provider

### PR DESCRIPTION
**Pull request by pierluigilenoci**
*Issue #, if available: #15

*Description of changes: Added a preliminary version of the chart that installs the software with the same configuration as the [deployment files](deployment/aws-provider-installer.yam) already present on the repo. Feel free to improve it or to suggest changes.

I put my name as maintainer because I did not know who else to insert but more than willingly I pass the burden and honors to someone else.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

This chart is a basic chart to replace point 3 of this guide: https://aws.amazon.com/fr/blogs/security/how-to-use-aws-secrets-configuration-provider-with-kubernetes-secrets-store-csi-driver/

### CODE REVIEW

**Comment by pierluigilenoci** [comment]
@joebaro could you please take a look?

**Comment by pierluigilenoci** [comment]
@joebaro @lasred @StevenEmelander could any of you take a look?

File: charts/csi-secrets-store-provider-aws/Chart.yaml
Comment (marcelobartsch) [inline charts/csi-secrets-store-provider-aws/Chart.yaml:1]: This should be v2 as helm v2 is already deprecated

File: charts/csi-secrets-store-provider-aws/templates/daemonset.yaml
Comment (marcelobartsch) [inline charts/csi-secrets-store-provider-aws/templates/daemonset.yaml:31]: I belive will make more sense to have aws instead of azure here

**Review by marcelobartsch — COMMENTED** [review]
Just a few comments, sorry don't know how to push commits to a PR so put them here.

File: charts/csi-secrets-store-provider-aws/Chart.yaml
Comment (pierluigilenoci) [reply charts/csi-secrets-store-provider-aws/Chart.yaml]: Done

File: charts/csi-secrets-store-provider-aws/templates/daemonset.yaml
Comment (pierluigilenoci) [reply charts/csi-secrets-store-provider-aws/templates/daemonset.yaml]: Done

Comment (pierluigilenoci) [reply charts/csi-secrets-store-provider-aws/templates/daemonset.yaml]: I work with both providers and sometimes it happens to get confused 😛

File: charts/csi-secrets-store-provider-aws/requirements.yaml
Comment (NissesSenap) [inline charts/csi-secrets-store-provider-aws/requirements.yaml:4]: There have been a 0.1.0 release, while you are at it please jump to the latest release.

**Review by NissesSenap — COMMENTED** [review]
Awesome job, looking forward to the release of this one :)

File: charts/csi-secrets-store-provider-aws/requirements.yaml
Comment (pierluigilenoci) [reply charts/csi-secrets-store-provider-aws/requirements.yaml]: Done

**Comment by pierluigilenoci** [comment]
@NissesSenap @marcelobartsch could you please take a look?

**Comment by NissesSenap** [comment]
I think it looks good, thanks for the bump.

**Comment by marcelobartsch** [comment]
agreed, looks good, now let's wait for a maintainer with write access to merge it!

**Review by marcelobartsch — APPROVED** [review]

**Comment by pierluigilenoci** [comment]
@NissesSenap could you please approve?

**Comment by pierluigilenoci** [comment]
@joebaro @lasred @StevenEmelander @willtong1234 @NissesSenap @parimaldeshmukh is there anyone who can help merge this PR?

**Comment by NissesSenap** [comment]
Not part of aws @pierluigilenoci , I'm just a happy user just like you are 👍

**Comment by felipeac** [comment]
LGTM as well.
Nice work, @pierluigilenoci

**Comment by pierluigilenoci** [comment]
Thank you @felipeac. But there is someone with write access that can approve the PR?

@joebaro @lasred @StevenEmelander could any of you take a look?

**Comment by palchyk-alex** [comment]
Hey @NissesSenap, can you approve this PR? :)

**Comment by NissesSenap** [comment]
@palchyk-alex not a part of AWS so sadly i can't. I waiting for someone at AWS to do this for a long time now.

**Comment by pierluigilenoci** [comment]
@palchyk-alex thanks for your effort. I tried in the past to ping someone from AWS without results. Let's try again. 😜 

@joebaro @lasred @StevenEmelander  could someone of you take a look?

The real problem is that this repo seems abandoned since August. 😢

**Comment by pierluigilenoci** [comment]
Maybe we can ping someone related to EKS... 🤔 
@maishsk @joebowbeer @kishorj @M00nF1sh @bwagner5 can you tell us who we can ask? 

Or maybe we are in the wrong place to open a PR.
Maybe it should be opened here: https://github.com/aws/eks-charts

**Comment by pierluigilenoci** [comment]
Ok, I opened an issue on the other side. 😜 
https://github.com/aws/eks-charts/issues/633

**Comment by pierluigilenoci** [comment]
Still no reaction... 😞

**Comment by bwagner5** [comment]
Apologies for the delay!! I will reach out to someone who is more familiar with this repo to review.

**Review by bwagner5 — COMMENTED** [review]
I'm not very deep into this repo but a few comments / questions.

File: charts/csi-secrets-store-provider-aws/values.yaml
Comment (pierluigilenoci) [reply charts/csi-secrets-store-provider-aws/values.yaml:7]: Not necessarily, they are opinions. However, I change it to IfNotPresent.

File: charts/csi-secrets-store-provider-aws/values.yaml
Comment (pierluigilenoci) [reply charts/csi-secrets-store-provider-aws/values.yaml:38]: They already have a default one here: https://github.com/pierluigilenoci/secrets-store-csi-driver-provider-aws/blob/helm_chart/charts/csi-secrets-store-provider-aws/templates/_helpers.tpl#L40
The values needs to have the `null` value.

File: charts/csi-secrets-store-provider-aws/templates/daemonset.yaml
Comment (pierluigilenoci) [reply charts/csi-secrets-store-provider-aws/templates/daemonset.yaml:40]: This chart contains also a Pod Security Policy:
https://github.com/pierluigilenoci/secrets-store-csi-driver-provider-aws/blob/helm_chart/charts/csi-secrets-store-provider-aws/templates/podsecuritypolicy.yaml 
This configuration says that the daemonset pod is authorized to use it:
https://kubernetes.io/docs/concepts/policy/pod-security-policy/#privileged

**Comment by pierluigilenoci** [comment]
> I'm not very deep into this repo but a few comments / questions.

@bwagner5 I made the suggested change and tried to clarify the rest of the doubts.
Let me know how to proceed.

**Comment by NissesSenap** [comment]
Now I will be a pain but while waiting for AWS to respond to this PR secrets-store-csi-driver have released 1.0.0 https://github.com/kubernetes-sigs/secrets-store-csi-driver/releases/tag/v1.0.0.

I don't know if this CSI driver is compatible with 1.0.0 and sadly I don't have time to try it out now but it would be nice to jump to 1.0.0 straight away.

They have also updated there helm chart location to: https://kubernetes-sigs.github.io/secrets-store-csi-driver/charts/index.yaml
as mentioned in: https://github.com/kubernetes-sigs/secrets-store-csi-driver/releases/tag/v1.0.0-rc.1

**Comment by pierluigilenoci** [comment]
@NissesSenap I'm working on it, I'm trying to do it for the beginning of next week.

**Comment by pierluigilenoci** [comment]
This PR should address these issues #15 #55 and it is also linked to this PR #56

**Comment by kitos9112** [comment]
Another AWS customer here uses the CSI driver to manage secrets within our EKS cluster. We'd be delighted to see this PR merged into main!

**Comment by pierluigilenoci** [comment]
@NissesSenap @bwagner5 could you please take a look?

**Comment by pierluigilenoci** [comment]
So, to get the PR approved, who do I have to bother? 😈

**Comment by pierluigilenoci** [comment]
@bwagner5 could you please take a look?
Maybe @joebaro or @lasred?

**Comment by pierluigilenoci** [comment]
@bwagner5 could you please take a look?

**Comment by pierluigilenoci** [comment]
@StevenEmelander could you please take a look?

**Comment by pierluigilenoci** [comment]
@willtong1234 @yuriadams @LeandroFranciscato @parimaldeshmukh could you please take a look?

**Comment by pierluigilenoci** [comment]
@kishorj @rexnp @saiteja313 @M00nF1sh @cgchinmay @shaileshgupta2k @achevuru could you please take a look?

File: charts/csi-secrets-store-provider-aws/README.md
Comment (bwagner5) [inline charts/csi-secrets-store-provider-aws/README.md:14]: let's make this eks-charts link

**Comment by pierluigilenoci** [comment]
As suggested by @bwagner5 I moved the PR here: https://github.com/aws/eks-charts/pull/651

When the new PR is merged I will close this too.

File: charts/csi-secrets-store-provider-aws/README.md
Comment (pierluigilenoci) [reply charts/csi-secrets-store-provider-aws/README.md]: Done. And moved here https://github.com/aws/eks-charts/pull/651

**Comment by pierluigilenoci** [comment]
@bwagner5 let me know if there is anything else I can do. 

Fast recap: 
* https://github.com/aws/eks-charts/issues/633 (an issue in EKS charts)
* https://github.com/aws/eks-charts/pull/651 (PR to merge the chart in EKS charts repo)
* https://github.com/aws/eks-charts/issues/354 (a related issue in EKS charts repo)
* https://github.com/aws/secrets-store-csi-driver-provider-aws/issues/15 (an issue in this repo)
* https://github.com/aws/secrets-store-csi-driver-provider-aws/pull/35 (this PR)
* https://github.com/aws/secrets-store-csi-driver-provider-aws/issues/55 (a related issue in this repo)
* https://github.com/aws/secrets-store-csi-driver-provider-aws/pull/56 (related PR in this repo)
* https://github.com/portefaix/portefaix-hub/pull/103 (related PR in another repo)

**Comment by pierluigilenoci** [comment]
The chart is finally merged on eks-charts repo. 
https://github.com/aws/eks-charts/tree/master/stable/csi-secrets-store-provider-aws

So this PR can be closed.

**Comment by NissesSenap** [comment]
Amazing job @pierluigilenoci!

**Pull request closed without merging**
